### ISSUE TITLE
A Classifier taking as input a dataset and some goals which output the method to use and its hyper-parameters

### ISSUE THREAD

**Issue by samuelgruffaz** [probabl-ai/skore#874]
### Is your feature request related to a problem? Please describe.

Users often make numerous attempts and comparisons with different models before selecting one. However, in theory, similar problems should yield similar solutions. Model selection can therefore be framed as a classification problem. Additionally, we may want to predict the AUC (Area Under the Curve) achievable with a selected model without the need for training.

### Describe the solution you'd like

An algorithm can be designed to take the data distribution and text-based requirements as inputs. The data distribution could be represented using a Gaussian mixture model and textual descriptions, while the text can be encoded into features using a large language model (LLM). Leveraging expert knowledge and a dataset consisting of (dataset, model used, scores) pairs, the algorithm can then predict the accuracy or AUC a user might achieve depending on the selected model.

### Describe alternatives you've considered, if relevant

This idea is interesting if the cost of prediction is less than the cost of training new models. 

### Additional context

Maybe in a first time, you can just try to predict the AUC of GradientBoosting tree depending on the dataset.

**Comment by sylvaincom** [comment]
Always great to hear your ideas @samuelgruffaz, thanks! This looks promising, requires some research, and could be an interesting new feature

**Comment by MarieSacksick** [comment]
Hello @samuelgruffaz, 
Thank you for this issue! It's an interesting question raised here: given a problem, what can we expect to be "performing good". I would argue that it's even a research topic where there is no clear way to answer this question yet. Therefore, I'll close this issue as not planned.  
Two leads though to get a hint of the performance obtained in a given problem before doing the whole exploration/training process: using a baseline, or using the [learning curves](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.learning_curve.html) to see the results on training with less data than the full dataset.
